lib/celluloid/io/stream.rb

```diff
@@ -80,6 +71,20 @@ def syswrite(string)
         total_written
       end
 
+      # TODO: remove after ending ruby 2.0.0 support
+      if RUBY_VERSION >= "2.1" 
+        def read_nonblock(*args, **options)
```

Comment:
This changes the semantics of this method on Ruby >= "2.1" making it exceptionless by default and unilaterally (i.e. even if `exception: true` were passed)

I'd rather we preserve Ruby core semantics as much as possible.

How about an internal `__read_nonblock` method which is always exceptional on Ruby < 2.1 and always exceptionless on 2.1+?